docs/modules/ROOT/pages/erc721.adoc

```diff
@@ -269,10 +262,7 @@ mod MyTokenReceiver {
 
 === Storing ERC721 URIs
 
-:string-roadmap: https://github.com/orgs/starkware-libs/projects/1/views/1?pane=issue&itemId=28823165[here]
+:erc721uri-storage: https://docs.openzeppelin.com/contracts/5.x/api/token/erc721#ERC721URIStorage[ERC721URIStorage]
 
-Token URIs in Cairo are stored as single field elements (`felt252`).
-Each field element equates to 252-bits (or 31.5 bytes) which means that a token's URI can be no longer than 31 characters.
-
-NOTE: Native string support in Cairo is currently in progress and tracked {string-roadmap}.
-Once Cairo offers full string support, this will be revisited.
+Token URIs were previously stored as single field elements prior to Cairo v0.2.5; however, they are now stored as ``ByteArray``s.
+This design is similar to {erc721uri-storage}.
```

Comment:
is it? we don't store the baseUri nor concatenate anything, [as the solidity module does](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v5.0.1/contracts/token/ERC721/extensions/ERC721URIStorage.sol#L46)

maybe we should consider that approach? we need to benchmark it

Reply:
Agreed. I'll look into it

Reply:
> maybe we should consider that approach? we need to benchmark it

At DX level, it's very interesting to only care about a `base_uri` set once, and have the automatic computation by appending the `token_id` when querying the `token_uri`.

```rust
# This is not constant in gas, the larger the token_id value, the more gas.
format!("{}{}", self.base_uri.read(), token_id)
```

Storage wise, it is way cheaper, no doubt.
Computation wise, if a transaction is used with a `token_uri` computation, using `format!` makes is more costly than reading it from the storage.

But is it a fair assumption to say that usually `token_uri` is called without transaction, hence without gas issues?